modules/pool/lib/pool-onchain-gyro-fee.ts

```diff
@@ -0,0 +1,86 @@
+import { Multicaller3 } from '../../web3/multicaller3';
+import { PrismaPoolType } from '@prisma/client';
+import abi from '../abi/GyroConfig.json';
+import { defaultAbiCoder } from '@ethersproject/abi';
+import { formatBytes32String } from '@ethersproject/strings';
+import { keccak256 } from '@ethersproject/solidity';
+import { formatEther } from 'ethers/lib/utils';
+
+interface PoolInput {
+    id: string;
+    address: string;
+    type: PrismaPoolType;
+    version?: number;
+}
+
+interface OnchainGyroFees {
+    eclpFee?: string;
+    defaultFee?: string;
+    pools?: {
+        [id: string]: {
+            poolFee?: string;
+        };
+    }
+}
+
+export const fetchOnChainGyroFees = async (
+  pools: PoolInput[],
+  gyroConfigAddress?: string,
+  batchSize = 1024
+) => {
+    if (pools.length === 0 || !gyroConfigAddress) {
+        return {};
+    }
+
+    const multicaller = new Multicaller3(abi, batchSize);
+
+    const feeKey = formatBytes32String('PROTOCOL_SWAP_FEE_PERC');
+
+    const eclpKey = keccak256(
+        ['bytes'],
+        [
+            defaultAbiCoder.encode(
+                ['bytes32', 'bytes32'],
+                [feeKey, formatBytes32String('ECLP')]
```

Comment:
Did you double check this key? any chance we can make this more general for other gyro pool types too?

Reply:
confirmed as ECLP. yes, adding other types as well!